utils/azureUtilityManager.ts

```diff
@@ -0,0 +1,138 @@
+        let registries: ContainerModels.Registry[] = [];
+
+        if (subscription && resourceGroup) {
+            //Get all registries under one resourcegroup
+            const client = this.getContainerRegistryManagementClient(subscription);
+            registries = await client.registries.listByResourceGroup(resourceGroup);
+
+        } else if (subscription) {
+            //Get all registries under one subscription
+            const client = this.getContainerRegistryManagementClient(subscription);
+            registries = await client.registries.list();
+
+        } else {
+            //Get all registries for all subscriptions
+            const subs: SubscriptionModels.Subscription[] = this.getFilteredSubscriptionList();
+            const subPool = new AsyncPool(MAX_CONCURRENT_SUBSCRIPTON_REQUESTS);
+
+            for (let sub of subs) {
+                subPool.addTask(async () => {
+                    const client = this.getContainerRegistryManagementClient(sub);
+                    let subscriptionRegistries: ContainerModels.Registry[] = await client.registries.list();
+                    registries = registries.concat(subscriptionRegistries);
+                });
+            }
+            await subPool.runAll();
+        }
+
+        if (sortFunction && registries.length > 1) {
+            registries.sort(sortFunction);
+        }
+
+        return registries;
+    }
+
+    public async getResourceGroups(subscription?: SubscriptionModels.Subscription): Promise<ResourceGroup[]> {
+        if (subscription) {
+            const resourceClient = this.getResourceManagementClient(subscription);
+            return await resourceClient.resourceGroups.list();
+        }
+        const subs = this.getFilteredSubscriptionList();
+        const subPool = new AsyncPool(MAX_CONCURRENT_SUBSCRIPTON_REQUESTS);
+        let resourceGroups: ResourceGroup[] = [];
+        //Acquire each subscription's data simultaneously
+        for (let sub of subs) {
+            subPool.addTask(async () => {
+                const resourceClient = this.getResourceManagementClient(sub);
+                const internalGroups = await resourceClient.resourceGroups.list();
+                resourceGroups = resourceGroups.concat(internalGroups);
+            });
+        }
+        await subPool.runAll();
+        return resourceGroups;
+    }
+
+    public getCredentialByTenantId(tenantId: string): ServiceClientCredentials {
+
+        const session = this.getAccount().sessions.find((azureSession) => azureSession.tenantId.toLowerCase() === tenantId.toLowerCase());
+
+        if (session) {
+            return session.credentials;
+        }
+
+        throw new Error(`Failed to get credentials, tenant ${tenantId} not found.`);
+    }
+
+    //CHECKS
+    //Provides a unified check for login that should be called once before using the rest of the singletons capabilities
+    public async isLoggedIn(): Promise<boolean> {
```

Comment:
Let's keep the concept count the same by naming it waitForLogin like AzureAccount.waitForLogin